skaffold/src/main/kotlin/com/google/container/tools/skaffold/run/SkaffoldFilesComboBox.kt

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.container.tools.skaffold.run
+
+import com.google.container.tools.skaffold.findSkaffoldFiles
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VfsUtilCore
+import com.intellij.openapi.vfs.VirtualFile
+import java.awt.Component
+import javax.swing.DefaultComboBoxModel
+import javax.swing.DefaultListCellRenderer
+import javax.swing.JComboBox
+import javax.swing.JLabel
+import javax.swing.JList
+import javax.swing.MutableComboBoxModel
+
+/**
+ * A combo box with a list of available Skaffold configuration files for a given project.
+ */
+class SkaffoldFilesComboBox : JComboBox<VirtualFile>() {
+    private lateinit var skaffoldFilesMutableModel: MutableComboBoxModel<VirtualFile>
+
+    /**
+     * Sets project for the combo box, and fills combo box with all Skaffold files available
+     * in this project. First file is selected by default.
+     */
+    fun setProject(project: Project) {
+        setRenderer(VirtualFileRenderer(project.baseDir))
```

Comment:
can you just do `renderer =` here? Same as how you are doing for model (and Kotlin implicitly calls the Java setter).

Reply:
Definitely, done. I wonder why IDEA didn't prompt me to, although it did on `model` access.

Reply:
thats interesting and slightly concerning that it didn't prompt you. It seems like it has the same behavior (by stepping into render it takes you to the setter). I wonder why it behaved differently.

Reply:
Reverted - didn't work properly. `setRenderer` calls Swing property listeners, this might be the cause -  I guess Kotlin style properties are only when you assign the value. Interesting to hear Kotlin reviewers opinion.

Reply:
hmm but I thought that if you call a Java class from Kotlin using the `render = `, then it will invoke the Java setter for render. I'm not sure I understand how render is different from model (the setter does a bunch of side-effect stuff there too). 

would be good to understand this (I probably just need to read about it again).